cmd/ingester/app/consumer/consumer_test.go

```diff
@@ -52,212 +49,283 @@ func TestConstructor(t *testing.T) {
+		"HeartbeatRequest": sarama.NewMockHeartbeatResponse(t),
+		"JoinGroupRequest": sarama.NewMockSequence(
+			sarama.NewMockJoinGroupResponse(t).SetError(sarama.ErrOffsetsLoadInProgress),
+			sarama.NewMockJoinGroupResponse(t).SetGroupProtocol(sarama.RangeBalanceStrategyName),
+		),
+		"SyncGroupRequest": sarama.NewMockSequence(
+			sarama.NewMockSyncGroupResponse(t).SetError(sarama.ErrOffsetsLoadInProgress),
+			sarama.NewMockSyncGroupResponse(t).SetMemberAssignment(
+				&sarama.ConsumerGroupMemberAssignment{
+					Version: 0,
+					Topics: map[string][]int32{
+						topic: {0},
+					},
+				}),
+		),
+		"OffsetFetchRequest": sarama.NewMockOffsetFetchResponse(t).
+			SetOffset(group, topic, 0, 0, "", sarama.ErrNoError).
+			SetOffset(group, topic, 0, 1, "", sarama.ErrNoError).
+			SetError(sarama.ErrNoError),
+		"FetchRequest": sarama.NewMockSequence(
+			sarama.NewMockFetchResponse(t, 1).
+				SetMessage(topic, 0, 0, sarama.StringEncoder(msg1)).
+				SetMessage(topic, 0, 1, sarama.StringEncoder(msg2)),
+			sarama.NewMockFetchResponse(t, 1),
+		),
+	})
 
-func TestSaramaConsumerWrapper_start_Messages(t *testing.T) {
-	localFactory := metricstest.NewFactory(0)
+	saramaConsumer, err := sarama.NewConsumerGroup([]string{broker0.Addr()}, group, config)
+	require.NoError(t, err)
 
-	msg := &sarama.ConsumerMessage{}
+	defer func() { _ = saramaConsumer.Close() }()
 
-	isProcessed := sync.WaitGroup{}
-	isProcessed.Add(1)
-	mp := &pmocks.SpanProcessor{}
-	mp.On("Process", saramaMessageWrapper{msg}).Return(func(msg processor.Message) error {
-		isProcessed.Done()
-		return nil
-	})
+	unmarshaller := kafka.NewJSONUnmarshaller()
+	innerSpanWriter := memory.NewStore()
 
-	saramaConsumer := smocks.NewConsumer(t, &sarama.Config{})
-	mc := saramaConsumer.ExpectConsumePartition(topic, partition, msgOffset)
-	mc.ExpectMessagesDrainedOnClose()
+	sw := &Store{
+		store:       innerSpanWriter,
+		spanWriteCh: make(chan struct{}, 1),
+	}
 
-	saramaPartitionConsumer, e := saramaConsumer.ConsumePartition(topic, partition, msgOffset)
-	require.NoError(t, e)
+	spParams := processor.SpanProcessorParams{
+		Writer:       sw,
+		Unmarshaller: unmarshaller,
+	}
 
-	undertest := newConsumer(t, localFactory, topic, mp, newSaramaClusterConsumer(saramaPartitionConsumer, mc))
+	spanProcessor := processor.NewSpanProcessor(spParams)
 
-	undertest.partitionIDToState = map[int32]*consumerState{
-		partition: {
-			partitionConsumer: &partitionConsumerWrapper{
-				topic:             topic,
-				partition:         partition,
-				PartitionConsumer: &kmocks.PartitionConsumer{},
-			},
-		},
+	logger, logBuf := testutils.NewLogger()
+	factoryParams := ProcessorFactoryParams{
+		Topic:          topic,
+		Parallelism:    1,
+		SaramaConsumer: saramaConsumer,
+		BaseProcessor:  spanProcessor,
+		Logger:         logger,
+		Factory:        metrics.NullFactory,
 	}
 
-	undertest.Start()
+	processorFactory, err := NewProcessorFactory(factoryParams)
+	assert.NoError(t, err)
```

Comment:
`processorFactory` will be invalid if `err != nil`.

```suggestion
	require.NoError(t, err)
```